Add documentation for keystore startup prompting (#50821)
elastic/elasticsearch: docs/reference/setup/secure-settings.asciidoc

NOTE: If you don't specify the `-p` flag or if you enter an empty password, the {es} keystore will be obfuscated but not password protected.

> Once the keystore is password-protected, Elasticsearch will prompt for a password at startup.

Reviewer: Slight variation to avoid future tense and to make it clear it's not a one-time prompt:
```suggestion
When the keystore is password-protected, {es} prompts you to supply the password each time it starts.
```

Author: Good suggestions!
